### PR TITLE
fix comparison of map fields

### PR DESCRIPTION
**Pull request by ekoby**

### CODE REVIEW

File: library/model_support.c [hunk 1]
```diff
@@ -86,24 +88,23 @@ int model_cmp(void *lh, void *rh, type_meta *meta) {
...
+            rf_ptr = rf_addr;
 
-            if (ftm->comparer) {
-                rc = ftm->comparer(lf_ptr, rf_ptr);
-            }
-            else {
-                rc = model_cmp(lf_ptr, rf_ptr, ftm);
-            }
+            rc = model_map_compare(lf_ptr, rf_ptr, ftm);
         }
```
Comment (dovholuknf) [inline library/model_support.c:106]: might be nice to add a catchall 'else' at the end for the next time? not that there'll be a next time? :)

File: library/model_support.c [hunk 2]
```diff
@@ -1009,6 +1010,43 @@ model_map_iter model_map_it_remove(model_map_iter it) {
     return next;
 }
 
+static int model_map_compare(model_map *lh, model_map *rh, type_meta *m) {
```
Comment (dovholuknf) [inline library/model_support.c:1013]: it feels like you really can't compare maps unless they're sorted, right? i don't think the map has a sort function does it?

**Review by dovholuknf — APPROVED** [review]

File: library/model_support.c (hunk 2 above)
Comment (ekoby) [reply library/model_support.c:1013]: sorting does not matter for comparison

Comment (dovholuknf) [reply library/model_support.c:1013]: not for the function to execute properly - sure. but for it to be accurate to what a human would expect it does.

Comment (ekoby) [reply library/model_support.c:1013]: a human would be wrong to make any assumptions wtr to map ordering

Comment (ekoby) [reply library/model_support.c:1013]: also this is an internal function, not part of the API

**Pull request merged**